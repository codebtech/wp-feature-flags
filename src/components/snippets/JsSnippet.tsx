import Snippet from './Snippet';
import { useMemo } from '@wordpress/element';
import { __ } from '@wordpress/i18n';
import Clipboard from '../common/Clipboard';

const JsSnippet = ({ flag }: { flag: string }) => {
	const jsSnippet = useMemo(() => {
		return `import domReady from '@wordpress/dom-ready';
domReady(function () {
	if (
		typeof window?.codebFeatureFlags !== 'undefined' &&
		window.codebFeatureFlags.isEnabled('${flag}')
	) {
		// js code goes here...
	}
});`;
	}, [flag]);
	return (
		<div className="mr-feature-flag-js-snippet-container">
			<h3>{__('JavaScript Snippet', 'codeb-feature-flags')}</h3>
			<Clipboard text={jsSnippet} />
			<Snippet data={jsSnippet} language={'typescript'} />
		</div>
	);
};

export default JsSnippet;
