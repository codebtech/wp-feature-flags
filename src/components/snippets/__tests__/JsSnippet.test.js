import JsSnippet from '../JsSnippet';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('JsSnippet component', () => {
	test('should render JsSnippet correctly with passed flag prop', async () => {
		render(<JsSnippet flag="testFlag" />);
		const result = screen.getByText(/JavaScript Snippet/i);
		expect(result).toBeInTheDocument();
	});

	test('should render the correct JavaScript snippet with the passed flag', async () => {
		render(<JsSnippet flag="testFlag" />);
		const snip = screen.getByText(/.codebFeatureFlags.isEnabled/i);
		expect(snip).toBeInTheDocument();
	});

	test('should update the JavaScript snippet when the flag prop changes', async () => {
		const { rerender, asFragment } = render(<JsSnippet flag="testFlag1" />);
		let snip = screen.getByText(/testFlag1'/i);
		expect(snip).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();

		rerender(<JsSnippet flag="testFlag2" />);
		snip = screen.queryByText(/'testFlag1'/i);
		expect(snip).toBeNull();

		snip = screen.getByText(/'testFlag2'/i);
		expect(snip).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});
});
