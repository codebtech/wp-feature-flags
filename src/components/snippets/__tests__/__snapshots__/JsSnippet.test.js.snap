// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsSnippet component should update the JavaScript snippet when the flag prop changes 1`] = `
<DocumentFragment>
  <div
    class="mr-feature-flag-js-snippet-container"
  >
    <h3>
      JavaScript Snippet
    </h3>
    <button
      aria-label="Copy to clipboard"
      aria-pressed="false"
      class="components-button codeb-feature-flags-clipboard-base is-tertiary has-icon"
      type="button"
    >
      <span
        class="dashicon dashicons dashicons-clipboard"
      />
    </button>
    <pre
      style="display: block; overflow-x: auto; background: rgb(43, 43, 43); color: rgb(248, 248, 242); padding: 0.5em;"
    >
      <code
        class="language-typescript"
        style="white-space: pre;"
      >
        <span
          style="color: rgb(220, 198, 224);"
        >
          import
        </span>
        <span>
           domReady 
        </span>
        <span
          style="color: rgb(220, 198, 224);"
        >
          from
        </span>
        <span>
           
        </span>
        <span
          style="color: rgb(171, 227, 56);"
        >
          '@wordpress/dom-ready'
        </span>
        <span>
          ;

        </span>
        <span>
          domReady(
        </span>
        <span
          class="hljs-function"
          style="color: rgb(220, 198, 224);"
        >
          function
        </span>
        <span
          class="hljs-function"
        >
           (
        </span>
        <span
          class="hljs-function"
        >
          ) 
        </span>
        <span>
          {

        </span>
        <span>
          	
        </span>
        <span
          style="color: rgb(220, 198, 224);"
        >
          if
        </span>
        <span>
           (

        </span>
        <span>
          		
        </span>
        <span
          style="color: rgb(220, 198, 224);"
        >
          typeof
        </span>
        <span>
           
        </span>
        <span
          style="color: rgb(245, 171, 53);"
        >
          window
        </span>
        <span>
          ?.codebFeatureFlags !== 
        </span>
        <span
          style="color: rgb(171, 227, 56);"
        >
          'undefined'
        </span>
        <span>
           &&

        </span>
        <span>
          		
        </span>
        <span
          style="color: rgb(245, 171, 53);"
        >
          window
        </span>
        <span>
          .codebFeatureFlags.isEnabled(
        </span>
        <span
          style="color: rgb(171, 227, 56);"
        >
          'testFlag1'
        </span>
        <span>
          )

        </span>
        	) {

        <span>
          		
        </span>
        <span
          style="color: rgb(212, 208, 171);"
        >
          // js code goes here...
        </span>
        <span>
          

        </span>
        	}
});
      </code>
    </pre>
  </div>
</DocumentFragment>
`;

exports[`JsSnippet component should update the JavaScript snippet when the flag prop changes 2`] = `
<DocumentFragment>
  <div
    class="mr-feature-flag-js-snippet-container"
  >
    <h3>
      JavaScript Snippet
    </h3>
    <button
      aria-label="Copy to clipboard"
      aria-pressed="false"
      class="components-button codeb-feature-flags-clipboard-base is-tertiary has-icon"
      type="button"
    >
      <span
        class="dashicon dashicons dashicons-clipboard"
      />
    </button>
    <pre
      style="display: block; overflow-x: auto; background: rgb(43, 43, 43); color: rgb(248, 248, 242); padding: 0.5em;"
    >
      <code
        class="language-typescript"
        style="white-space: pre;"
      >
        <span
          style="color: rgb(220, 198, 224);"
        >
          import
        </span>
        <span>
           domReady 
        </span>
        <span
          style="color: rgb(220, 198, 224);"
        >
          from
        </span>
        <span>
           
        </span>
        <span
          style="color: rgb(171, 227, 56);"
        >
          '@wordpress/dom-ready'
        </span>
        <span>
          ;

        </span>
        <span>
          domReady(
        </span>
        <span
          class="hljs-function"
          style="color: rgb(220, 198, 224);"
        >
          function
        </span>
        <span
          class="hljs-function"
        >
           (
        </span>
        <span
          class="hljs-function"
        >
          ) 
        </span>
        <span>
          {

        </span>
        <span>
          	
        </span>
        <span
          style="color: rgb(220, 198, 224);"
        >
          if
        </span>
        <span>
           (

        </span>
        <span>
          		
        </span>
        <span
          style="color: rgb(220, 198, 224);"
        >
          typeof
        </span>
        <span>
           
        </span>
        <span
          style="color: rgb(245, 171, 53);"
        >
          window
        </span>
        <span>
          ?.codebFeatureFlags !== 
        </span>
        <span
          style="color: rgb(171, 227, 56);"
        >
          'undefined'
        </span>
        <span>
           &&

        </span>
        <span>
          		
        </span>
        <span
          style="color: rgb(245, 171, 53);"
        >
          window
        </span>
        <span>
          .codebFeatureFlags.isEnabled(
        </span>
        <span
          style="color: rgb(171, 227, 56);"
        >
          'testFlag2'
        </span>
        <span>
          )

        </span>
        	) {

        <span>
          		
        </span>
        <span
          style="color: rgb(212, 208, 171);"
        >
          // js code goes here...
        </span>
        <span>
          

        </span>
        	}
});
      </code>
    </pre>
  </div>
</DocumentFragment>
`;
